ext/opcache/ZendAccelerator.c

```diff
@@ -1162,13 +1167,32 @@ zend_result validate_timestamp_and_record(zend_persistent_script *persistent_scr
 {
 	if (persistent_script->timestamp == 0) {
 		return SUCCESS; /* Don't check timestamps of preloaded scripts */
-	} else if (ZCG(accel_directives).revalidate_freq &&
-	    persistent_script->dynamic_members.revalidate >= ZCG(request_time)) {
+	}
+
+	double revalidate_reference_time = 0.0;
+
+	if (ZCG(cli_mode)) {
```

Comment:
https://www.php.net/manual/en/function.php-sapi-name.php - maybe there is a better way to detect cli /wo extra property.

Also the time getter can be probably deduplicated to something like `microtime(true)`.

Reply:
Regarding `microtime(true)`, I believe there may be some confusion here. The `microtime()` function referenced is a userland PHP function and not directly available within the engine context.

The implementation I used is based on the internal `php_time()` logic, with slight adjustments to return a double for better precision.

I also avoided using `microtime()` as-is because it's not available consistently across all platforms.

If there’s a preferred helper or abstraction for obtaining a double-precision current time in the engine, I’m happy to switch to that but I wanted to avoid introducing inconsistencies across platforms.

Reply:
Anyway, it looks like this part of PR may be weak, and I'm happy for any suggestions to it. Maybe I can drop double precision logic and switch to `php_time()` here.